test/arithmetic/fixed_to_float_test.dart

```diff
@@ -0,0 +1,166 @@
+    }
+  });
+
+  test('Unsigned Q4.4 to E3M2', () async {
+    final fixed = FixedPoint(signed: false, m: 4, n: 4);
+    final dut = FixedToFloat(fixed, exponentWidth: 3, mantissaWidth: 2);
+    await dut.build();
+    for (var val = 0; val < pow(2, fixed.width); val++) {
+      final fixedValue = FixedPointValue(
+          value: LogicValue.ofInt(val, fixed.width),
+          signed: fixed.signed,
+          m: fixed.m,
+          n: fixed.n);
+      fixed.put(fixedValue);
+      final fpv = dut.float.floatingPointValue;
+      final fpvExpected = FloatingPointValue.ofDouble(fixedValue.toDouble(),
+          exponentWidth: dut.exponentWidth, mantissaWidth: dut.mantissaWidth);
+      expect(fpv.sign, fpvExpected.sign);
+      expect(fpv.exponent.bitString, fpvExpected.exponent.bitString,
+          reason: 'exponent mismatch');
+      expect(fpv.mantissa.bitString, fpvExpected.mantissa.bitString,
+          reason: 'mantissa mismatch');
+    }
+  });
+
+  test('Signed Q0.8 to E3M2 shrink', () async {
+    final fixed = FixedPoint(signed: true, m: 0, n: 7);
+    final dut = FixedToFloat(fixed, exponentWidth: 3, mantissaWidth: 2);
+    await dut.build();
+    for (var val = 0; val < pow(2, fixed.width); val++) {
+      final fixedValue = FixedPointValue(
+          value: LogicValue.ofInt(val, fixed.width),
+          signed: fixed.signed,
+          m: fixed.m,
+          n: fixed.n);
+      fixed.put(fixedValue);
+      final fpv = dut.float.floatingPointValue;
+      final fpvExpected = FloatingPointValue.ofDouble(fixedValue.toDouble(),
+          exponentWidth: dut.exponentWidth, mantissaWidth: dut.mantissaWidth);
+      expect(fpv.sign, fpvExpected.sign);
+      expect(fpv.exponent.bitString, fpvExpected.exponent.bitString,
+          reason: 'exponent mismatch');
+      expect(fpv.mantissa.bitString, fpvExpected.mantissa.bitString,
+          reason: 'mantissa mismatch');
+    }
+  });
+
+  test('Signed Q0.3 to E5M6 expand', () async {
+    final fixed = FixedPoint(signed: true, m: 0, n: 3);
+    final dut = FixedToFloat(fixed, exponentWidth: 5, mantissaWidth: 6);
+    await dut.build();
+    for (var val = 0; val < pow(2, fixed.width); val++) {
+      final fixedValue = FixedPointValue(
+          value: LogicValue.ofInt(val, fixed.width),
+          signed: fixed.signed,
+          m: fixed.m,
+          n: fixed.n);
+      fixed.put(fixedValue);
+      final fpv = dut.float.floatingPointValue;
+      final fpvExpected = FloatingPointValue.ofDouble(fixedValue.toDouble(),
+          exponentWidth: dut.exponentWidth, mantissaWidth: dut.mantissaWidth);
+      expect(fpv.sign, fpvExpected.sign);
+      expect(fpv.exponent.bitString, fpvExpected.exponent.bitString,
+          reason: 'exponent mismatch');
+      expect(fpv.mantissa.bitString, fpvExpected.mantissa.bitString,
+          reason: 'mantissa mismatch');
+    }
+  });
+
+  // Test is skipped as FloatingPointValue.ofDouble does not handle infinities.
```

Comment:
you could mark this as a `// TODO(owner): ... <link to bug>` so that we know to enable the test once the issue is fixed.  it's also worth adding a reminder in the github issue so whenever it is fixed we remember to enable this